src/submit.rs

```diff
@@ -476,4 +477,70 @@ impl<'a> Submitter<'a> {
         )
         .map(drop)
     }
+
+    /// Performs a synchronous cancellation request, similiar to [AsyncCancel](crate::opcode::AsyncCancel),
+    /// except that it completes synchronously.
+    ///
+    /// Cancelation can target a specific request, or all requests matching a specific criteria. The
+    /// [MatchOn](types::MatchOn) builder supports describing the match criteria for cancelation.
+    ///
+    /// An optional `timeout` can be provided to specify how long to wait for matched requests to be
+    /// canceled. If no timeout is provided, the default is to wait indefinitely.
+    ///
+    /// ### Errors
+    ///
+    /// An error is returned if no requests match the provided match criteria. If a timeout is supplied,
+    /// and the timeout expires before all requests have been canceled, then an error will be returned.
```

Comment:
Given that a timeout return would be important to recognize, I think it useful to call out that it returns ETIME (or whatever the io::Error module converts that too).